lib/readResources.js

```diff
@@ -95,7 +95,7 @@ function readResource(gltf, object, options) {
     // Source already exists if the gltf was converted from a glb
     var source = object.extras._pipeline.source;
     if (defined(source)) {
-        return Promise.resolve(source);
+        return Promise.resolve(Buffer.from(source));
```

Comment:
The problem I was running into was that when running these tests, it was a JSON object contain the data, and not a full `Buffer` object. I could add a check here so it only does the conversion if it's not already an object.

Reply:
OK, I'll take another look :+1:

Reply:
The issue I'm running into when running the tests is that `parseGlb` sets this to a `TypedArray` rather than a `Buffer`.

Should I just convert it to a `Buffer`? Or is that not OK since we want to use this in Cesium as well?